sessionctx/variable/sysvar_test.go

```diff
@@ -834,3 +836,113 @@ func TestTiDBServerMemoryLimit(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, strconv.FormatUint(200*mb, 10), val)
 }
+
+func TestTiDBServerMemoryLimit2(t *testing.T) {
+	vars := NewSessionVars(nil)
+	mock := NewMockGlobalAccessor4Tests()
+	mock.SessionVars = vars
+	vars.GlobalVarsAccessor = mock
+	var (
+		//mb  uint64 = 1 << 20
+		err error
+		val string
+	)
+	// Test tidb_server_memory_limit
+	serverMemoryLimit := GetSysVar(TiDBServerMemoryLimit)
+	// Check default value
+	require.Equal(t, serverMemoryLimit.Value, DefTiDBServerMemoryLimit)
+
+	total := memory.GetMemTotalIgnoreErr()
+	if total > 0 {
+		// Can use percentage format when TiDB can obtain physical memory
+		// Test Percentage Format
+		err = mock.SetGlobalSysVar(context.Background(), TiDBServerMemoryLimit, "1%")
+		require.NoError(t, err)
+		val, err = mock.GetGlobalSysVar(TiDBServerMemoryLimit)
+		require.NoError(t, err)
+		require.Equal(t, memory.ServerMemoryLimit.Load(), uint64(512<<20))
+		require.Equal(t, "512MB", val)
```

Comment:
Maybe it is a good idea to assert `512MB` first. I am not sure if `1%` will always return 512..